add hls method to write ts files

AndreaNicola opened the pull request:
Greetings! I needed to create some m3u8 playlists. I added an Hls method to Video struct that inserts the necessary options for ffmpeg to do them.

AndreaNicola committed:
add hls method to write ts files

AndreaNicola commented:
Hello! :-D is there anybody out there? :-D

hixi commented on moviego.go:
```diff
@@ -88,6 +90,13 @@ func (V Video) ResizeByWidth(requiredWidth int64) Video {
 
+func (V Video) Hls() Video {
+	V.addKwArgs("f", "hls")
+	V.addKwArgs("hls_time", "3")
```
Default for this is 2 in ffmpeg, is there a specific reason to set this to 3?

AndreaNicola replied:
i don't think so... maybe i could add a parameter to Hls method and overwrite the default only if necessary!

hixi replied:
For me that would be clearer to have the same behavior as when using ffmpeg directly (see https://ffmpeg.org/ffmpeg-formats.html#Options-26).

AndreaNicola committed:
Hls method with default hls_tim

added a method to specify hls with hlstime and removed hls_time option from Hls method to use default setting

AndreaNicola commented:
@hixi Hi! i modified the method to use default hls_time and added a new method to set hls custom time :-)

hixi reviewed (approved):
Looks good to me. Thank you :-)

@mowshon what do you think?

AndreaNicola commented:
lol
💀

mowshon commented:
The new version of the library is now available. This PR was taken into account during the refactoring of the package.

@AndreaNicola  You can now export the playlist this way.

@hixi You can now specify any custom settings for FFmpeg in `ExtraOutputArgs`.

```golang
mgo.WriteVideo(ctx, clip, "out.m3u8", mgo.ExportOptions{
    Rate: mgo.Rate{Num: 30, Den: 1},
    ExtraOutputArgs: []string{
        "-f", "hls",
        "-hls_time", "3",
    },
})
```
